File: 4.ConvertStringToLowercase/PYTHON/ConvertStringToLowercase.py
# Python program to convert a string to lowercase
# Visit https://venkys.io for more information

class ConvertStringToLowercase:
    def convert_to_lowercase(self, input_string):
        result = ''
        for char in input_string:
            if 'A' <= char <= 'Z':
                result += chr(ord(char) + 32)
            else:
                result += char
        return result

# Input from the user
input_string = input("Please enter a string: ")

# Create an instance of the ConvertStringToLowercase class
converter = ConvertStringToLowercase()

# Call the function to convert the string to lowercase
lowercase_string = converter.convert_to_lowercase(input_string)

# Display the original and converted strings
print("\nOriginal: ", input_string)
print("Converted: ", lowercase_string)

File: 4.ConvertStringToLowercase/JAVA/src/ConvertStringToLowercase.java
import java.util.Scanner;
public class ConvertStringToLowercase {
    // Function to convert a string to lowercase
    public static String toLowerCase(String input) {
        StringBuilder result = new StringBuilder();
        for (char ch : input.toCharArray()) {
            if (ch >= 'A' && ch <= 'Z') {
                result.append((char) (ch + 32));
            } else {
                result.append(ch);
            }
        }
        return result.toString();
    }

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.print("Enter the string to convert to lowercase: ");
            String input = scanner.nextLine();
            String lowercase = toLowerCase(input);
            System.out.println("Lowercase: " + lowercase);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
